### PR TITLE
Handle empty files in `read_xsv` & friends

### PR DESCRIPTION
**Pull request by IyeOnline**

### CODE REVIEW

File: libtenzir/builtins/formats/xsv.cpp [hunk 1]
```diff
@@ -633,6 +633,10 @@ auto parse_loop(generator<std::optional<std::string_view>> lines,
                 xsv_parser_options args) -> generator<table_slice> {
   // Parse header.
   auto it = lines.begin();
+  if (it == lines.end()) {
+    co_yield {};
+    co_return;
+  }
```
Comment (jachris) [inline libtenzir/builtins/formats/xsv.cpp:639]: I don't think that works, in particular because of our empty-thing-means-yield.

File: changelog/next/bug-fixes/5215--emtpy_xsv.md [hunk 2]
```diff
@@ -0,0 +1,2 @@
+We fixed a bug that caused `read_xsv` & friends to crash when trying to read an
+empty file.
```
Comment (dominiklohmann) [inline changelog/next/bug-fixes/5215--emtpy_xsv.md:2]: Can you add a test? Something along the lines of `from env("INPUTS") + "/csv/empty.csv"`

Comment (IyeOnline) [reply changelog/next/bug-fixes/5215--emtpy_xsv.md:2]: Great minds think alike! :)

File: libtenzir/builtins/formats/xsv.cpp (hunk 1 above)
Comment (dominiklohmann) [reply libtenzir/builtins/formats/xsv.cpp]: I don't quite understand this comment. I think the `co_yield {}` is unnecessary, but other than that this looks good?

Comment (jachris) [reply libtenzir/builtins/formats/xsv.cpp]: These 4 lines were located at the top of the function when I commented this. Now that they are moved it should work as intended, I agree.
